api/models/post/util.js

```diff
@@ -172,3 +172,32 @@ export const createThread = (userId, params) =>
       pick(params, 'messageTo'),
       {children: params.requests, transacting: trx}
     )))))
+
+export const addFollowers = (post, comment, userIds, addedById, opts = {}) => {
+  var userId = (comment || post).get('user_id')
+  const { transacting, createActivity } = opts
+
+  return post.load(['communities'])
+  .then(() => post.isProject() && post.load('selectedTags')
+    .then(() => {
+      const tag = post.relations.selectedTags.first()
+      if (!tag) return
+      return Promise.each(post.relations.communities.models, community =>
+        Promise.each(userIds, id => TagFollow.add(tag, id, community)))
+    }))
+  .then(() => Promise.map(userIds, followerId =>
+    Follow.create(followerId, post.id, (comment || {}).id, {addedById, transacting})
+    .tap(follow => {
+      if (!createActivity) return
+
+      var updates = []
+      const addActivity = (recipientId, method) => {
+        updates.push(Activity[method](follow, recipientId)
+        .save({}, _.pick(opts, 'transacting'))
+        .then(activity => activity.createNotifications(transacting)))
+      }
+      if (followerId !== addedById) addActivity(followerId, 'forFollowAdd')
```

Comment:
agreed, and hallelujah. I'll create a new task for this because it touches a few places